pkg/bufferdecoder/eventsreader.go

```diff
@@ -66,67 +34,66 @@ func readArgFromBuff(id events.ID, ebpfMsgDecoder *EbpfDecoder, params []trace.A
 		return 0, arg, errfmt.Errorf("invalid arg index %d", argIdx)
 	}
 	arg.ArgMeta = params[argIdx]
-	argType := GetParamType(arg.Type)
+	argType := GetDecodeType(arg.Type)
 
 	switch argType {
-	case u8T:
+	case trace.U8_T:
```

Comment:
How about reducing the size of `readArgFromBuff()` by replacing the switch/if branches with an already initialized array containing pointer functions for each type (ebpfMsgDecoder.Decode*)?

Reply:
This is basically the planned next step. I plan to move the decoding mapping to a hash map similar to the processor and derive functions. In this way we, and plugin authors, can easily introduce new decoding strategies.

Reply:
@geyslan I've opted not to make this step in this PR but rather in a followup so that we merge smaller chunks at a time as this is a large scale change.